BenmoussaYounes/chat-app
# 10- featurre integrate conversation API and add chat missing functionalities ✨

## JiraTicket [FI-10](https://contributors.atlassian.net/jira/software/projects/ATX/boards/6?selectedIssue=ATX-10)

## Description 📑
- integrate conversation api page functionality and add needed functionality to chat page to make the app features works correctly

## Type of Change


## Changes

This pull request includes the following changes:
- integrate conversation api including send, receive functionality
- adding new getLastMessage API to chat to show latest updates according to conversation
- add user enum values, fromString func
- add conversation bloc, repos to di, and remove unnecessary abstraction repos used for chat
- add seen status to message bubble

## Screenshots 📷

| 01                                    | 02                                    | 03                                    | 04                                    |
| ------------------------------------- | ------------------------------------- | ------------------------------------- | ------------------------------------- |
| <img src="https://github.com/user-attachments/assets/0964604c-4ae0-418c-b7a4-1ecf656db27e" width="100" height="250"> | <img src="https://github.com/user-attachments/assets/7554a630-5fd4-4931-b84a-5595d883b34f" width="100" height="250"> | <img src="https://github.com/user-attachments/assets/17726bcc-1ab0-47a6-9d27-cabdd91cbc20" width="100" height="250"> | <img src="https://github.com/user-attachments/assets/da8a69f6-a7a7-4785-9239-bfb59330efd6" width="100" height="250"> |

## How to test 🚦

- run the app, send messages between ali and younes users and check if the ui is updating correctly
